src/components/FormElements/ContractMethodSelectPicker.tsx

```diff
@@ -7,29 +7,51 @@ import { Box } from "@/components/layout/Box";
 import { JsonSchemaForm } from "@/components/JsonSchemaForm";
 import { ExpandBox } from "@/components/ExpandBox";
 
+import { SorobanInvokeValue } from "@/types/types";
+
 // @todo for testing
 // comment it out before committing
 // remove it before merging into main
 import Form from "@rjsf/core";
 import validator from "@rjsf/validator-ajv8";
+
+const cleanupData = (data: any) => {
+  if (!data) return {};
+  // Only keep function_method if it exists, reset all other fields
+  return {
+    function_method: data.function_method || "",
+  };
+};
+
 export const ContractMethodSelectPicker = ({
+  value,
   methods,
   spec,
   id,
+  onChange,
 }: {
+  value: SorobanInvokeValue;
   methods: string[];
   spec: contract.Spec;
   id: string;
+  onChange: (val: SorobanInvokeValue) => void;
 }) => {
   const [selectedValue, setSelectedValue] = useState<string>("");
   const [funcSpec, setFuncSpec] = useState<JSONSchema7 | undefined>(undefined);
 
-  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    setSelectedValue(e.target.value);
-
+  const onSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
     if (e.target.value) {
       const selectedFuncSchema = spec.jsonSchema(e.target.value);
+      setSelectedValue(e.target.value);
       setFuncSpec(selectedFuncSchema);
+
+      onChange({
+        ...value,
+        data: {
+          ...cleanupData(value.data),
```

Comment:
clean up its stored data (fields) whenever function method gets updated in zustand store

Reply:
isn't it safer to make this cleanup right after the request and before setting the store? This way you ensure that other places that may eventually use this won't get a different data structure.